### DISCUSSION TITLE
Propagating property change to parent object

### DISCUSSION THREAD

**Discussion by JkPersson** [dotnet/maui#16539] [Q&A]
I would like to utilize 'dot-notation' in my views to avoid dummy properties in the view models. However I have not been able to propagate UI changes of child properties to the view model. Is it even possible? Following is a simplified example of what I am trying to accomplish:

View:
`<?xml version="1.0" encoding="utf-8" ?>
<ContentPage
    x:Class="MauiApp1.Views.MyPage"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:viewmodels="clr-namespace:MauiApp1.ViewModels"
    Title="MyPage"
    x:DataType="viewmodels:MyViewModel">
    <VerticalStackLayout>
        <Entry Text="{Binding A.B.Bb}" />
        <Button IsEnabled="{Binding IsButtonEnabled}" Text="{Binding ButtonText}"/>
        
        <Label Text="{Binding A.B.Bbb}" />
        <Label Text="{Binding A.Aa}" />
    </VerticalStackLayout>
</ContentPage>`

ViewModel:
`using CommunityToolkit.Mvvm.ComponentModel;
using MauiApp1.Models;

namespace MauiApp1.ViewModels;
internal partial class MyViewModel : ObservableObject
{
    internal MyViewModel()
    {
        a = new A { B = new B() };
    }

    [ObservableProperty]
    private A a;

    partial void OnAChanged(A value)
    {
        // In the real world implementation, I would have some validation code here.
        IsButtonEnabled = true;
        ButtonText = "Enabled";
    }

    [ObservableProperty]
    private bool isButtonEnabled = false;

    [ObservableProperty]
    private string buttonText = "Disabled";
}
`

Model classes:
`using CommunityToolkit.Mvvm.ComponentModel;

namespace MauiApp1.Models;
internal partial class A : ObservableObject
{
    [ObservableProperty]
    private B b;
    partial void OnBChanged(B value)
    {
        Aa = "Aa: " + value.Bbb;
    }

    [ObservableProperty]
    private string aa;
}

using CommunityToolkit.Mvvm.ComponentModel;

namespace MauiApp1.Models;
internal partial class B : ObservableObject
{
    [ObservableProperty]
    private string bb;
    partial void OnBbChanged(string value)
    {
        Bbb = "Bbb: " + value;
    }

    [ObservableProperty]
    private string bbb;
}
`

Label Bbb is updating just fine, but Aa and the button is not. 
Also A.OnBChanged is not called - which is probably the core of my problem. Please help.

**Comment by Mzazvor** [comment]
Hey did you manage to solve this ?

**Comment by stephenquan** [comment]
In A's view model, you can subscribe to B's PropertyChanged event handler, e.g.

```c#
internal partial class A : ObservableObject
{
    public B B { get; } = new B();
    [ObservableProperty]
    private string aa;
    public A()
    {
        B.PropertyChanged += (s, e) =>
        {
            switch (e.PropertyName)
            {
                case nameof(B.Bbb):
                    Aa = "Aa: " + value.Bbb;
                    break;
            }
        };
    }
}
```

Because Aa is a derivative of B.Bbb, I would implement Aa with a read-only getter, i.e.

```c#
internal partial class A : ObservableObject
{
    public B B { get; } = new B();
    public string Aa => "Aa: " + B.Bbb;
    public A()
    {
        B.PropertyChanged += (s, e) =>
        {
            switch (e.PropertyName)
            {
                case nameof(B.Bbb):
                    OnPropertyChanged(nameof(Aa));
                    break;
            }
        };
    }
}
```
